core/src/main/java/hudson/model/Fingerprint.java

```diff
@@ -1234,74 +1251,28 @@ public int compare(FingerprintFacet o1, FingerprintFacet o2) {
     }
 
     /**
-     * Save the settings to a file.
+     * Save the Fingerprint in the Fingerprint Storage
      * @throws IOException Save error
      */
     public synchronized void save() throws IOException {
-        if(BulkChange.contains(this))   return;
+        if(BulkChange.contains(this)) return;
 
         long start=0;
         if(logger.isLoggable(Level.FINE))
             start = System.currentTimeMillis();
 
-        File file = getFingerprintFile(md5sum);
-        save(file);
-        SaveableListener.fireOnChange(this, getConfigFile(file));
+        FingerprintStorage.get().save(this);
 
         if(logger.isLoggable(Level.FINE))
-            logger.fine("Saving fingerprint "+file+" took "+(System.currentTimeMillis()-start)+"ms");
+            logger.fine("Saving fingerprint took "+(System.currentTimeMillis()-start)+"ms");
     }
 
+    /**
+     * Save the Fingerprint in the given file locally
+     * @throws IOException Save error
+     */
     void save(File file) throws IOException {
```

Comment:
It is not a public method though it can be accessed from the same package in plugins. Do you see examples of that? 

* If yes, then it makes sense to deprecate the method
* If not, we could consider removing it entirely

Reply:
@oleg-nenashev I can see examples of it being used in the tests. So to make sure I understand you correctly, I should remove dependence on this method, and then mark it as deprecated?